### PR TITLE
Setup Project

### PR DESCRIPTION
**Pull request by Efiamotu-1**
### Project requirements Done
 * Create a new Rails app called 'hello-rails-react'.
   
   * Use the command `rails new -j webpack my_new_rails_app` as described in the previous lesson's tutorial.
 * Verify that the `jsbundling-rails` gem is added to your Gemfile.
 * Initialize your project with Git.
 * Install needed node dependencies as described in the previous lesson's tutorial.
 * Create a `babel.config.js` in the root folder of your project as described in the previous lesson's tutorial.
 * Update `webpack.config.js` to include the babel loader as described in the previous lesson's tutorial.
 * Make sure that your project has a Postgres database set up.
 * Create a table for storing your messages and create 5 different greetings.
 * Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
 * Create a static view that will be the root of your app.
 
 * Create the `App` component with react-router and render it in your static view.
 * Create the `Greeting` component that will display a greeting. Set it up as a route in your App component.
 * Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
 * Display the random greeting in your `Greeting` component.

### CODE REVIEW

File: MIT.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-## Copyright 2021, [YOUR NAME]
+## Copyright 2021, [HABEEB EFIAMOTU MUSA OWOLEWA]
```
Comment (Davidosky007) [inline MIT.md:1]: * Kindly create 5 random greetings as mentioned in the project [requirements ](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md#:~:text=create%20a%20table%20for%20storing%20your%20messages%20and%20create%205%20different%20greetings.)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

* No linter errors 
* Well written PR description 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: MIT.md (hunk 1 above)
Comment (Efiamotu-1) [reply MIT.md:1]: I have created the greetings and they are all displaying. @Davidosky007

File: app/javascript/pages/Greeting.js [hunk 2]
```diff
@@ -6,16 +6,16 @@ const Greeting = () => {
...
     dispatch(fetchGreeting());
   }, []);
 
   return (
-    <div>{Object.values(greeting).map((greeting) => (
-      <h1>{greeting.message}</h1>
-  ))}
-  </div>
-    )
+    greeting?.map(greet => {
```
Comment (rustam-tolipov) [inline app/javascript/pages/Greeting.js:15]: [OPTIONAL] You completed the previous reviewer's requirement well. However, your app should generate random greetings and display them in the front end when we refresh the page.

**Review by rustam-tolipov — APPROVED** [review]
Hi @Efiamotu-1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯

## Optional suggestions
_Kindly check comments below_

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/pages/Greeting.js (hunk 2 above)
Comment (Efiamotu-1) [reply app/javascript/pages/Greeting.js:15]: @Rustamxon7  can you give morer insights  on this optional requiremennt? where was this made and i couln't find a link to your project to even see how you did yours. i am waiting to hear back from you

Comment (rustam-tolipov) [reply app/javascript/pages/Greeting.js:15]: Hi @Efiamotu-1, I will leave the student's project link. [The link](https://github.com/Camilovelag/hello-rails-react/blob/dev/app/controllers/api/v1/greetings_controller.rb).

**Pull request merged**